src/components/Layout/Layout.js

```diff
@@ -1,41 +1,110 @@
-import React, { Component } from 'react';
+import React from 'react';
 
 import SEO from '../SEO';
 import Flex from '../Flex';
 import Footer from '../LayoutFooter';
 import Header from '../LayoutHeader';
+import HamburgerMenu from './../Hamburger/Menu';
+
+import { media } from '../theme';
+
+class Layout extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      menuToggle: false,
+      hasScroll: false
+    };
+    this.mql = null;
```

Comment:
what is the mql?

Reply:
it's an object to handle the window.matchMedia() function. 'mql' for media query listener.

Reply:
so then maybe this.mediaQueryListener = null; :)

Reply:
done